### PR TITLE
feat(DataTable + DataGrid): hideSelectAll, scrollToTopOnUpdate, conditional horizontal scroll shadow, headers overlap fix, docs update

### PR DESCRIPTION
**Pull request by dreamwasp**
### Overview

* hideSelectAll added to DataTable
* fixes Column header overlapping in Tables
* scrollToTopOnUpdate prop added to allow scroll to top on component update
* removed right side shadow on tables when scrolled all the way to the right or when no scroll is available

[MONOLITH](https://github.com/codecademy-engineering/Codecademy/pull/30482)
[PORTAL](https://github.com/codecademy-engineering/portal-app/pull/1886)
[BAM](https://github.com/codecademy-engineering/business-account-management/pull/623)

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: web-1944
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.102-alpha.f03cd7.0<br/>@codecademy/gamut-labs@31.7.0-alpha.f03cd7.0<br/>@codecademy/gamut@46.2.0-alpha.f03cd7.0<br/>@codecademy/monaco@0.6.34-alpha.f03cd7.0<br/>@codecademy/styleguide@56.4.1-alpha.f03cd7.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62598232b7c5590089ba6f85--gamut-preview.netlify.app>https://62598232b7c5590089ba6f85--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62598232b7c5590089ba6f85>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.102-alpha.f7014f.0<br/>@codecademy/gamut-labs@31.7.0-alpha.f7014f.0<br/>@codecademy/gamut@46.2.0-alpha.f7014f.0<br/>@codecademy/monaco@0.6.34-alpha.f7014f.0<br/>@codecademy/styleguide@56.4.1-alpha.f7014f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6259eb96803dea05d0a08622--gamut-preview.netlify.app>https://6259eb96803dea05d0a08622--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6259eb96803dea05d0a08622>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.107-alpha.a35944.0<br/>@codecademy/gamut-labs@31.9.0-alpha.a35944.0<br/>@codecademy/gamut@46.2.0-alpha.a35944.0<br/>@codecademy/monaco@0.6.35-alpha.a35944.0<br/>@codecademy/styleguide@56.4.6-alpha.a35944.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62619d579e43522e7f4e94af--gamut-preview.netlify.app>https://62619d579e43522e7f4e94af--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62619d579e43522e7f4e94af>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.111-alpha.864630.0<br/>@codecademy/gamut-labs@31.11.0-alpha.864630.0<br/>@codecademy/gamut@46.2.0-alpha.864630.0<br/>@codecademy/monaco@0.6.36-alpha.864630.0<br/>@codecademy/styleguide@56.4.10-alpha.864630.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6261c906a81b45534785d609--gamut-preview.netlify.app>https://6261c906a81b45534785d609--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6261c906a81b45534785d609>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.111-alpha.a58a22.0<br/>@codecademy/gamut-labs@31.11.0-alpha.a58a22.0<br/>@codecademy/gamut@46.2.0-alpha.a58a22.0<br/>@codecademy/monaco@0.6.36-alpha.a58a22.0<br/>@codecademy/styleguide@56.4.10-alpha.a58a22.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62670725e47d25077808bdf2--gamut-preview.netlify.app>https://62670725e47d25077808bdf2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62670725e47d25077808bdf2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.114-alpha.0dfba9.0<br/>@codecademy/gamut-labs@31.11.3-alpha.0dfba9.0<br/>@codecademy/gamut@46.2.0-alpha.0dfba9.0<br/>@codecademy/monaco@0.6.37-alpha.0dfba9.0<br/>@codecademy/styleguide@56.4.13-alpha.0dfba9.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.114-alpha.7d11a3.0<br/>@codecademy/gamut-labs@31.11.3-alpha.7d11a3.0<br/>@codecademy/gamut@46.2.0-alpha.7d11a3.0<br/>@codecademy/monaco@0.6.37-alpha.7d11a3.0<br/>@codecademy/styleguide@56.4.13-alpha.7d11a3.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.115-alpha.1b21ca.0<br/>@codecademy/gamut-labs@31.11.4-alpha.1b21ca.0<br/>@codecademy/gamut@47.1.0-alpha.1b21ca.0<br/>@codecademy/monaco@0.6.38-alpha.1b21ca.0<br/>@codecademy/styleguide@57.0.1-alpha.1b21ca.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.116-alpha.f353b5.0<br/>@codecademy/gamut-labs@31.11.5-alpha.f353b5.0<br/>@codecademy/gamut@47.1.0-alpha.f353b5.0<br/>@codecademy/monaco@0.6.39-alpha.f353b5.0<br/>@codecademy/styleguide@57.0.2-alpha.f353b5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62687de03e199d11a0146bf1--gamut-preview.netlify.app>https://62687de03e199d11a0146bf1--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62687de03e199d11a0146bf1>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.116-alpha.a0550b.0<br/>@codecademy/gamut-labs@31.11.5-alpha.a0550b.0<br/>@codecademy/gamut@47.1.0-alpha.a0550b.0<br/>@codecademy/monaco@0.6.39-alpha.a0550b.0<br/>@codecademy/styleguide@57.0.2-alpha.a0550b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://626be66bb19f29082450df45--gamut-preview.netlify.app>https://626be66bb19f29082450df45--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/626be66bb19f29082450df45>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://626fe4a8ce71d2470771ab90--gamut-preview.netlify.app>https://626fe4a8ce71d2470771ab90--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/626fe4a8ce71d2470771ab90>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.118-alpha.a58cf4.0<br/>@codecademy/gamut-labs@31.11.7-alpha.a58cf4.0<br/>@codecademy/gamut@47.2.0-alpha.a58cf4.0<br/>@codecademy/monaco@0.6.40-alpha.a58cf4.0<br/>@codecademy/styleguide@57.0.6-alpha.a58cf4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://626fe9d41e7a674994e05590--gamut-preview.netlify.app>https://626fe9d41e7a674994e05590--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/626fe9d41e7a674994e05590>Deploy Logs</a></p>

File: packages/gamut/src/DataList/Rows/HeaderRow.tsx [hunk 1]
```diff
@@ -37,21 +40,27 @@ export const Header: HeaderComponent = ({
...
       )}
       {columns.map(({ key, header, sortable, filters, ...colProps }) => {
         const rowProperty = key as string;
         const renderKey = prefixId(`header-col-${rowProperty}`);
-        const columnText = header || key;
+        const columnText = (
+          <Text truncate="ellipsis" truncateLines={5}>
+            {header || key}
+          </Text>
+        );
```
Comment (dreamwasp) [inline packages/gamut/src/DataList/Rows/HeaderRow.tsx:63]: i thought truncating longer column header would be the best move but open to other solutions @songstephen

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.118-alpha.4a3371.0<br/>@codecademy/gamut-labs@31.11.7-alpha.4a3371.0<br/>@codecademy/gamut@47.2.0-alpha.4a3371.0<br/>@codecademy/monaco@0.6.40-alpha.4a3371.0<br/>@codecademy/styleguide@57.0.6-alpha.4a3371.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://626fedcdce71d24f81720fd9--gamut-preview.netlify.app>https://626fedcdce71d24f81720fd9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/626fedcdce71d24f81720fd9>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.119-alpha.f8c82e.0<br/>@codecademy/gamut-labs@31.11.8-alpha.f8c82e.0<br/>@codecademy/gamut@47.2.0-alpha.f8c82e.0<br/>@codecademy/monaco@0.6.40-alpha.f8c82e.0<br/>@codecademy/styleguide@57.0.7-alpha.f8c82e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6271348757e2f60075a28d20--gamut-preview.netlify.app>https://6271348757e2f60075a28d20--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6271348757e2f60075a28d20>Deploy Logs</a></p>

File: packages/styleguide/stories/Organisms/Lists & Tables/DataList.stories.mdx [hunk 2]
```diff
@@ -152,6 +152,11 @@ const DataTown = () => {
 
 <br />
 
+### Additional Column Configuration
+
+- `size` - enum which determines how much space the column should take up within the table
+- `fill` - boolean which, when set to true, takes up the remaining space in the table. Each `DataTable` should only have one `fill` column.
```
Comment (tchryssos) [inline packages/styleguide/stories/Organisms/Lists & Tables/DataList.stories.mdx:158]: Is this true? I actually set a few things to be fill and it seems to behave pretty well. Things that are set to a fixed size are that size, and the remaining size is split between all the fill items.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.119-alpha.6039c8.0<br/>@codecademy/gamut-labs@31.11.8-alpha.6039c8.0<br/>@codecademy/gamut@47.2.0-alpha.6039c8.0<br/>@codecademy/monaco@0.6.40-alpha.6039c8.0<br/>@codecademy/styleguide@57.0.7-alpha.6039c8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62713a8816b5ce08aef99b7e--gamut-preview.netlify.app>https://62713a8816b5ce08aef99b7e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62713a8816b5ce08aef99b7e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.119-alpha.4adefe.0<br/>@codecademy/gamut-labs@31.11.8-alpha.4adefe.0<br/>@codecademy/gamut@47.2.0-alpha.4adefe.0<br/>@codecademy/monaco@0.6.40-alpha.4adefe.0<br/>@codecademy/styleguide@57.0.7-alpha.4adefe.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6272ea30c551cd0086f4742d--gamut-preview.netlify.app>https://6272ea30c551cd0086f4742d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6272ea30c551cd0086f4742d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.119-alpha.852e24.0<br/>@codecademy/gamut-labs@31.11.8-alpha.852e24.0<br/>@codecademy/gamut@47.2.0-alpha.852e24.0<br/>@codecademy/monaco@0.6.40-alpha.852e24.0<br/>@codecademy/styleguide@57.0.7-alpha.852e24.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62741feb4e2c0b00589a81cf--gamut-preview.netlify.app>https://62741feb4e2c0b00589a81cf--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62741feb4e2c0b00589a81cf>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.119-alpha.8407c4.0<br/>@codecademy/gamut-labs@31.11.8-alpha.8407c4.0<br/>@codecademy/gamut@47.2.0-alpha.8407c4.0<br/>@codecademy/monaco@0.6.40-alpha.8407c4.0<br/>@codecademy/styleguide@57.0.7-alpha.8407c4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62744a45263898101da06e86--gamut-preview.netlify.app>https://62744a45263898101da06e86--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62744a45263898101da06e86>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.119-alpha.5d0ed2.0<br/>@codecademy/gamut-labs@31.11.8-alpha.5d0ed2.0<br/>@codecademy/gamut@47.2.0-alpha.5d0ed2.0<br/>@codecademy/monaco@0.6.40-alpha.5d0ed2.0<br/>@codecademy/styleguide@57.0.7-alpha.5d0ed2.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62756677ebf46a242ccf2d36--gamut-preview.netlify.app>https://62756677ebf46a242ccf2d36--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62756677ebf46a242ccf2d36>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.119-alpha.a0ba4a.0<br/>@codecademy/gamut-labs@31.11.8-alpha.a0ba4a.0<br/>@codecademy/gamut@47.2.0-alpha.a0ba4a.0<br/>@codecademy/monaco@0.6.40-alpha.a0ba4a.0<br/>@codecademy/styleguide@57.0.7-alpha.a0ba4a.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.119-alpha.4f7974.0<br/>@codecademy/gamut-labs@31.11.8-alpha.4f7974.0<br/>@codecademy/gamut@47.2.0-alpha.4f7974.0<br/>@codecademy/monaco@0.6.40-alpha.4f7974.0<br/>@codecademy/styleguide@57.0.7-alpha.4f7974.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6275807aebf46a3c72cf2f6b--gamut-preview.netlify.app>https://6275807aebf46a3c72cf2f6b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6275807aebf46a3c72cf2f6b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.122-alpha.7a911a.0<br/>@codecademy/gamut-labs@31.11.11-alpha.7a911a.0<br/>@codecademy/gamut@47.2.0-alpha.7a911a.0<br/>@codecademy/monaco@0.6.41-alpha.7a911a.0<br/>@codecademy/styleguide@57.0.11-alpha.7a911a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627585e999edd83ed4d1f10a--gamut-preview.netlify.app>https://627585e999edd83ed4d1f10a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627585e999edd83ed4d1f10a>Deploy Logs</a></p>

**Review by CharlieBliss — APPROVED** [review]

**Review by vonbarown — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.122-alpha.19d938.0<br/>@codecademy/gamut-labs@31.11.11-alpha.19d938.0<br/>@codecademy/gamut@47.2.0-alpha.19d938.0<br/>@codecademy/monaco@0.6.41-alpha.19d938.0<br/>@codecademy/styleguide@57.0.11-alpha.19d938.0<br/></pre>

**Pull request merged**